File: Java/5.18.java
/*
Write a program that reads an integer, a list of words, and a character. The integer signifies how many words are in the list. The output of the program is every word in the list that contains the character at least once. Assume at least one word in the list will contain the given character. Assume that the list of words will always contain fewer than 20 words.

Ex: If the input is:

4 hello zoo sleep drizzle z
then the output is:

zoo
drizzle
To achieve the above, first read the list into an array. Keep in mind that the character 'a' is not equal to the character 'A'.
*/

import java.util.Scanner;

public class LabProgram {
   public static void main(String[] args) {
      /* Type your code here. */
      Scanner scan = new Scanner(System.in);
      String[] userInputs = new String[20];
      char userCheck;
      int arrSize = scan.nextInt();
      
      for (int i = 0; i < arrSize; i++) {
         userInputs[i] = scan.next();
      }
      
      userCheck = scan.next().charAt(0);
      
      for (int i = 0; i < arrSize; i++) {
         for (int j = 0; j < userInputs[i].length(); j++) {
            if (userInputs[i].charAt(j) == userCheck) {
               if (userInputs[j].equals(userInputs[j])) {
                  System.out.printf("%s\n", userInputs[i]);
               }
            }
         }
      }
   }
}

File: CPP/5.18.cpp
/*
Write a program that reads a list of integers, and outputs those integers in reverse. The input begins with an integer indicating the number of integers that follow. For coding simplicity, follow each output integer by a comma, including the last one.

Ex: If the input is:

5 2 4 6 8 10
the output is:

10,8,6,4,2,
To achieve the above, first read the integers into a vector. Then output the vector in reverse.
*/

#include <iostream>
#include <vector>   // Must include vector library to use vectors
using namespace std;

int main() {
   vector<int> userInts; // A vector to hold the user's input integers

   /* Type your code here. */
   int totalInts;
   cin >> totalInts;
   
   userInts.resize(totalInts);
   
   for (int i = 0; i < userInts.size(); i++) {
      cin >> userInts.at(i);
   }
   
   for (int i = userInts.size() - 1; i >= 0; i--) {
      printf("%d,", userInts.at(i));
   }
   cout << endl;

   return 0;
}
